src/utils/upcoming-matches.ts

```diff
@@ -0,0 +1,19 @@
+import { ProcessedMatchInfo } from '@/api/match-info'
+// import { useCurrentTime } from './use-current-time'
+
+export const getUpcomingMatches = (
+  matches: ProcessedMatchInfo[],
+  currentTime: number,
+) => {
+  const retArr = []
+  for (const match of matches) {
+    if (
+      match.time &&
+      match.time.getTime() < currentTime + 1000 * 60 * 15 &&
+      match.time.getTime() > currentTime - 1000 * 60 * 4
+    ) {
+      retArr.push(match)
+    }
+  }
+  return retArr
```

Comment:
@varCepheid can you try using the `.filter` method for arrays here? It simplifies the code a little bit and makes it easier to see what is going on:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

It would look something like:


```js
return matches.filter((match) => match.time && match.time.getTime......)
```

Then you don't have to deal with creating the return array yourself and pushing items to it, the `filter` method is made to do exactly that automatically!